yoyodyne/models/lstm.py

```diff
@@ -239,21 +244,27 @@ def decode(
                 decoder_input, decoder_hiddens, encoder_out, encoder_mask
             )
             predictions.append(output.squeeze(1))
-            # If we have a target (training) then the next input is the gold
-            # symbol for this step (i.e., teacher forcing).
-            if target is not None:
+            # In teacher forcing mode the next input is the gold symbol
+            # for this step.
+            if teacher_forcing:
                 decoder_input = target[:, t].unsqueeze(1)
```

Comment:
I have a minor suggestion. Put an assertion (or module-specific error) here to catch when `teacher_forcing` is True but `target` is None. It shouldn't be reachable but if it is that'd be better than `TypeError: 'NoneType' object is not subscriptable`.

```python
if teacher_forcing:
   assert target is not None, "Teacher forcing requested but no target provided"
   ...
```